### DISCUSSION TITLE
bounds vs set_nl_cons

### DISCUSSION THREAD

**Discussion by SR-dTdt** [do-mpc/do-mpc#223] [Q&A]
Hello,

I am trying to set hard lower and upper bounds on an algebraic state ("_z").

The bounds are fully respected if I am using "bounds" for the lower and upper bound of z.

However, using the command set_nl_cons (soft_constraint=False) leads to the phenomena, that bound are not respected. 

Have you faced this issue before? 

I am using the following software version:
1. casadi==3.5.5
2. do-mpc==4.1.1

I am using the mumps solver so far.

Best regards,
DTdt

**Comment by 4flixt** [comment]
Hi, thank you posting your issue. That is indeed strange. The bounds should be met exactly with the flag ``soft_constraints=False``. Would you be able to share a (more or less) minimal example of the problem?
I will investigate asap :)

Best
4flixt
